src/litdata/streaming/downloader.py

```diff
@@ -26,8 +27,17 @@
 
 class Downloader(ABC):
     def __init__(
-        self, remote_dir: str, cache_dir: str, chunks: List[Dict[str, Any]], storage_options: Optional[Dict] = {}
+        self,
+        cloud_provider: str,
+        remote_dir: str,
+        cache_dir: str,
+        chunks: List[Dict[str, Any]],
+        storage_options: Optional[Dict] = {},
     ):
+        print("-" * 100)
+        print(f"{cloud_provider=}")
+        print("-" * 100)
+        self.fs = fsspec.filesystem(cloud_provider)
```

Comment:
How do you provide the storage options to fsspec ?

Reply:
To be honest, I don't know much about storage options.

I'll look into s3 & gs docs to see different storage options available. And then will look back into fsspec docs to see how to pass them.

Reply:
https://filesystem-spec.readthedocs.io/en/latest/api.html#fsspec.filesystem

![Screenshot from 2024-09-02 19-07-45](https://github.com/user-attachments/assets/8525358d-bbbb-4fe4-8ca9-3d666cbf9ae6)

Probably this one.

Reply:
It works. But, there's slight mismatch in storage options and what it expects. Independent of our code, but user need to be aware of the `storage_options`.

```python
storage_options = {
    "key": "MY_AWS_ACCESS_KEY",
    # "aws_access_key_id": "MY_AWS_ACCESS_KEY",

   "secret": "MY_AWS_SECRET_KEY",
   #  "aws_secret_access_key": "MY_AWS_SECRET_KEY",
}
```

expects `key` instead of `aws_access_key_id` and `secret` instead of `aws_secret_access_key`.

<details>
  <summary>Complete code</summary>
  
  ```python
import litdata as ld

s3_uri = 's3://my-dummy-bucket-litdata/my_litdata/'

storage_options = {
    "key": "MY_AWS_ACCESS_KEY",
    # "aws_access_key_id": "MY_AWS_ACCESS_KEY",

   "secret": "MY_AWS_SECRET_KEY",
   #  "aws_secret_access_key": "MY_AWS_SECRET_KEY",
}

dataset = ld.StreamingDataset(s3_uri, shuffle=True, storage_options=storage_options)
dataloader = ld.StreamingDataLoader(dataset)

for sample in dataloader:
    print(sample)
    print("="*80)
print("\nAll done\n")

  ```
</details>

---

IMO, this happens bcoz of the way the [S3FS class, init function & set_session works](https://github.com/fsspec/s3fs/blob/2763a083e55d64abded49269daf098be86b4a961/s3fs/core.py#L276)